File: tools/improve/tests/gemini-3-flash/maudec/cpp/collatz.cpp
//
//	<comment from="gemini-3-flash-preview">
//	The implementation is simplified by using implicit conversion for `std::optional` to reduce verbosity and keeping the conditional logic concise. The code preserves the original semantics, which specifically handle even integers and positive odd integers, while returning `nullopt` for negative odd integers (where `term % 2` results in `-1`).
//	</comment>
//

#include <optional>

std::optional<int> rlapp_Nat(int term) {
    if (term % 2 == 0) {
        return term / 2;
    }
    if (term % 2 == 1) {
        return 3 * term + 1;
    }
    return std::nullopt;
}

std::optional<int> arlapp_Nat(int term) {
    return rlapp_Nat(term);
}

File: tools/improve/tests/gemini-3-flash/maudec/py/collatz.py
#
#	<comment from="gemini-3-flash-preview">
#	The implementation was simplified by replacing the match-case statement with a more efficient conditional expression. The redundant 'case _' was removed because an integer is always either even or odd. The semantics remain identical, but the code is more concise and has less execution overhead.
#	</comment>
#

def rlapp_Nat(term: int) -> int | None:
    return term // 2 if term % 2 == 0 else 3 * term + 1

def arlapp_Nat(term: int) -> int | None:
    return rlapp_Nat(term)